lib/Providers/GameProvider.dart

```diff
@@ -47,21 +47,58 @@ class GameProvider extends ChangeNotifier {
     notifyListeners();
   }
 
-  void playTurn() async {
+  void playTurn(BuildContext context) async {
     // Play Dice Roll Animation
     if (_currentEntity != null) {
-      int pRoll = _diceManager.roll(player.selectedDie);
+      int pRoll = _diceManager.roll(_player.selectedDie);
       int eRoll = _diceManager.roll(_currentEntity.selectedDie);
-      print("Player Rolled $pRoll");
-      _damageEntity(pRoll);
-      await new Future.delayed(const Duration(seconds: 1));
-      lastRoll = pRoll.toString();
-      // Player roll, current selected die
-      // Play Attack Animation
-      // Should reduce entities health etc etc
-      print("Entity Rolled $eRoll");
-      _damagePlayer(eRoll);
-      gameMessage = "${currentSelectedEntity.name} rolled a $eRoll";
+      // Don't allow any actions if the enemy is already dead
+      // This is mostly as a first line of defense to prevent the reward popup showing multiple times
+      // But also prevents a dead enemy from still dealing damage
+      if (_currentEntity.health != 0) {
+        print("Player Rolled $pRoll");
+        // _damageEntity(pRoll);
+        _damageEntity(100);
+        // If the enemy was killed just now, show the reward
+        // Must be placed here, or else the animation runtime will allow for the reward popup to
+        //   display several times if the Attack button is spammed
+        if (_currentEntity.health <= 0) {
+          showRewardAlert(context);
```

Comment:
similar to here, what's probably better, we should instead have a modal item in game provider that holds the current modal item, and a property called isModalOpen or something of that nature that is watched, then we can dynamically show the modal without having to pass around context and the like, since a watched variable will rebuild the widget when the property changes.